.github/workflows/pr-validation.yml

```diff
@@ -20,8 +20,30 @@ jobs:
       - name: Build devcontainer image
         run: docker build --tag devcontainer:1.0 .devcontainer
 
+      - name: Run devcontainer image
+        id: devcontainer
+        run: |
+          container_id=$(docker create -w /workspace -v $GITHUB_WORKSPACE:/workspace -v /var/run/docker.sock:/var/run/docker.sock devcontainer:1.0)
+          docker start "$container_id"
+          echo "::set-output name=id::$container_id"
+        
       - name: Run CI tasks
-        run: docker run -w /workspace -v $GITHUB_WORKSPACE:/workspace -e AZURE_TENANT_ID -e AZURE_CLIENT_ID -e AZURE_CLIENT_SECRET -e AZURE_SUBSCRIPTION_ID devcontainer:1.0 task ci
+        run: |
+          docker exec "${{steps.devcontainer.outputs.id}}" task ci
+
+      - name: Build docker image
+        run: |
+          container_id=${{steps.devcontainer.outputs.id}}
+          docker exec "$container_id" task controller:docker-build
+          docker cp "$container_id:/workspace/hack/generated/bin" .
+          mv bin/*.tar azure-service-operator.tar
+
+      - name: Archive Docker image
+        uses: actions/upload-artifact@v2
+        with:
```

Comment:
In the ADO job we publish to an MCR feed instead of uploading to artifacts. Do we want to do something similar here?

Reply:
This is just for the PR validation run, so you can download the output from it. We’ll add another workflow for the actual release.